File: JPLearningHub_Server/src/JPLearningHub_Server/Models/StudentModel.cs
using JPLearningHub_Server.Entities;

namespace JPLearningHub_Server.Models
{
    public class StudentModel
    {
        public UserModel User { get; set; } = new();
        public long Credits { get; set; }

        public static StudentModel FromStudent(Student student, string avatarUrl)
        {
            return new StudentModel
            {
                User = UserModel.FromUser(student, avatarUrl),
                Credits = student.Credits
            };
        }
    }
}

File: JPLearningHub_Client/lib/models/student_model.dart
import 'package:jplearninghub/models/user_model.dart';
import 'package:json_annotation/json_annotation.dart';

part 'student_model.g.dart';

@JsonSerializable()
class StudentModel {
  UserModel user;
  int credits;

  StudentModel({required this.user, required this.credits});

  Map<String, dynamic> toJson() => _$StudentModelToJson(this);
  factory StudentModel.fromJson(Map<String, dynamic> json) =>
      _$StudentModelFromJson(json);
}
